src/core/mac/mac.hpp

```diff
@@ -101,6 +101,139 @@ enum
     kIndirectFrameMacTxAttempts = OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_PER_POLL,
 };
 
+/**
+ * This class defines a channel mask.
+ *
+ * It is a wrapper class around a `uint32_t` bit vector representing a set of channels.
+ *
+ */
+class ChannelMask
```

Comment:
Would it make sense to generalize the Mask utilities and tests and just subclass it?
Also may want to consider splitting this out into a separate file.

Reply:
We talked about that already and decided to defer that for later.